### DISCUSSION TITLE
About FiraCode monospace variant

### DISCUSSION THREAD

**Discussion by polirritmico** [ryanoasis/nerd-fonts#1471] [Q&A] [answered]
Hi, I have had this question for a long time, but now that I am updating to version `3.1.1` I take the opportunity to ask:

Why does the **FiraCode** font have both a `FiraCodeNerdFont` and a `FiraCodeNerdFontMono` variant in the zipfile? Upon checking the FiraCode GitHub repo, I noticed that only FiraCode is provided and it is already monospaced.
Whats the difference between them beyond the file size? Which one should I use?

Thanks!

**Comment by Finii** [comment, marked as answer]
Sorry, yes that (the differences) are a bit under-documented.

In fact there are three variants
* `Nerd Font`
* `Nerd Font Mono`
* `Nerd Font Propo`

The first variant without a special addition/modifier can be called `Regular` maybe.

The modifier describes the added icons, not the glyphs of the original font (at least not mainly, more on that later).
With `Mono` the added icons have the same size as the rest of the font. So the Github icon is as wide as the capital letter `M`.

Some people feel the icons are too small if they are just as wide as the biggest ordinary letters - especially because they are often less high than wide and need to be scaled far down. For these people the `Regular` font has bigger icons. The icons are usually/often/sometimes around 1.6 'widths' wide. This makes them better readable.
But of course they occupy space that the next letter would need, so usually you need to have a blank after an icon to prevent overwriting.

And then there are people who want bigger icons but NO blank afterwards. Or they use the font in not-terminal contexts, for these the `Propo` variant is usually best. Here the icons are wider but do really occupy the visual space - the font is not monospaced anymore.

See for example (ordered with decreasing significance)
1. #1103  
1. #969
1. #951

To answer the concrete question. That depends in which environment you want to use the font. In terminals and programming environments usually `Regular` (the variant without a special suffix) is usually best. If your terminal / IDE can not work with that font you need to use `Mono` instead. If you use the font to create slides or for a status indicator, the `Propo` is usually the easiest to use.

---

The `Mono` variant is always completely monospaced (i.e. all glyphs have the same width), even if the source font is not. That is the only variant that touches (the width of) the original glyphs. A `Mono` font shall work on all that picky terminals that need really really monospaced fonts. Some original fonts that say they are "Mono" are in fact not (most prominent/newest example is `Monaspace`). The Nerd Font patcher 'repairs' that fonts so that they are also usable with the most picky terminals, i.e. work there, which the original does not.

_Edit: Fix some typos_

**Reply by polirritmico** [reply]
Thank you very much for such a comprehensive response. The criteria for choosing one or the other are now really clear to me.

Thanks!
